pkg/controller/azure/storage/container/container.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright 2018 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"context"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+
+	"github.com/crossplaneio/crossplane/pkg/clients/azure"
+
+	"github.com/crossplaneio/crossplane/pkg/util"
+
+	"github.com/crossplaneio/crossplane/pkg/clients/azure/storage"
+
+	corev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/crossplaneio/crossplane/pkg/apis/azure/storage/v1alpha1"
+)
+
+const (
+	controllerName = "account.storage.azure.crossplane.io"
+	finalizer      = "finalizer." + controllerName
+
+	reconcileTimeout      = 2 * time.Minute
+	requeueAfterOnSuccess = 1 * time.Minute
+
+	// failed condition reasons
+	//
+	// failed to get resource handler
+	failedToGetHandler = "error getting handler"
+	// failed to delete account resource
+	failedToDelete = "error deleting"
+	// failed to retrieve account resource
+	failedToRetrieve = "error retrieving"
+	// failed to create account resource
+	failedToCreate = "error creating"
+	// failed to update account resource
+	failedToUpdate = "error updating"
+)
+
+var (
+	resultRequeue    = reconcile.Result{Requeue: true}
+	requeueOnSuccess = reconcile.Result{RequeueAfter: requeueAfterOnSuccess}
+)
+
+// Reconciler reconciles a GCP storage account acct
+type Reconciler struct {
+	client.Client
+	syncdeleterMaker
+}
+
+// Add creates a newSyncdeleter Controller and adds it to the Manager with default RBAC.
+// The Manager will set fields on the Controller and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	r := &Reconciler{
+		Client:           mgr.GetClient(),
+		syncdeleterMaker: &containerSyncdeleterMaker{mgr.GetClient()},
+	}
+
+	// Create a newSyncdeleter controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to storage account
```

Comment:
this is watching for changes to container not to account, right?

Reply:
That is correct. Updated.